### PR TITLE
Adding documentation to State Sync code

### PR DESCRIPTION
**Pull request by mm-near**
Adding documentation to StateSync code and removed state_sync_time that was not used.

### CODE REVIEW

File: chain/client/src/sync/state.rs [hunk 1]
```diff
@@ -194,10 +217,14 @@ impl StateSync {
             let mut this_done = false;
             match &shard_sync_download.status {
                 ShardSyncStatus::StateDownloadHeader => {
+                    // StateDownloadHeader is the first step. We want to fetch the basic information about the state (its size, hash etc).
+                    // FIXME: why are we looking only on the first index? - ah right, we assume taht download header has only 1.
```
Comment (nikurt) [inline chain/client/src/sync/state.rs:221]:
```suggestion
                    // FIXME: why are we looking only on the first index? - ah right, we assume that download header has only 1.
```

Comment (nikurt) [inline chain/client/src/sync/state.rs:221]: Not clear what this FIXME requires to fix

File: chain/client/src/sync/state.rs [hunk 2]
```diff
@@ -247,6 +277,7 @@ impl StateSync {
                             }
                         }
                     }
+                    // If all parts are done - we can move towards scheduling (TODO: why?)
```
Comment (nikurt) [inline chain/client/src/sync/state.rs:280]: This TODO is redundant given another TODO at line 293.

**Review by nikurt — APPROVED** [review]

File: chain/client/src/sync/state.rs (hunk 1 above)
Comment (mm-near) [reply chain/client/src/sync/state.rs]: removed.

File: chain/client/src/sync/state.rs (hunk 2 above)
Comment (mm-near) [reply chain/client/src/sync/state.rs:280]: good point. done.

**Pull request merged**